### ISSUE TITLE
jest/valid-describe false positive for descibe.each variant

### ISSUE THREAD

**Issue by topaxi** [jest-community/eslint-plugin-jest#334]
The rule `jest/valid-describe` errors for the `describe.each` variant:

```javascript
describe.each([...data])('Something', data => {
  // ...
})
```

leads to:

```
  3:15  error  First argument must be name                    jest/valid-describe
  3:15  error  Describe requires name and callback arguments  jest/valid-describe
```

**Comment by joshuadiablito** [comment]
Also have the same error, it's valid: https://jestjs.io/docs/en/api#describeeachtable-name-fn-timeout

**Comment by G-Rath** [comment]
I suspect I know what's causing this, but I can't reproduce this - I managed to have it show up once in my IDE, but it didn't have a stacktrace and went away after a second.

I hate to be a pain, but if you could provide a small repo that reproduced the error, or even a stacktrace, that'd be a great help!

**Comment by SimenB** [comment]
Pasting in the snippet in the OP fails for me.

Diff with failing test:

```diff
diff --git i/src/rules/__tests__/valid-describe.test.ts w/src/rules/__tests__/valid-describe.test.ts
index adfbad5..5f63b49 100644
--- i/src/rules/__tests__/valid-describe.test.ts
+++ w/src/rules/__tests__/valid-describe.test.ts
@@ -37,6 +37,11 @@ ruleTester.run('valid-describe', rule, {
       test('bar', () => {})
     )
     `,
+    `
+    describe.each([...data])('Something', data => {
+      // ...
+    })
+    `,
   ],
   invalid: [
     {
```

**Comment by G-Rath** [comment]
eeekkkk but totally what I expected - I'll try and wipe up a fix asap.

**Comment by SimenB** [comment]
The _real_ fix is to handle `describe.each`. The _quick_ fix is to not recognize it as a `describe` - `each` is pretty different, so I think really fixing it is a bit more work than needed

**Comment by G-Rath** [comment]
One step ahead of you ;)

I'm going to make a quick fix to ignore `each`, and open a new issue to support `each`.

The rule readme actually excludes `each`:

>The following describe function aliases are also validated:
>
>describe
>describe.only
>describe.skip
>fdescribe
>xdescribe

**Comment by G-Rath** [comment]
Sorry my quick fix ended up not being so quick as I got distracted finishing off the conversion for `prefer-todo` :joy:

**Comment by topaxi** [comment]
Wow! Thanks for the quick fix! <3

**Comment by topaxi** [comment]
For some reason, this now triggers on completely unrelated code:

```typescript
import { hasOwnProperty } from './has-own-property'

export function size(obj: object): number {
  let size = 0

  for (let key in obj) {
    if (hasOwnProperty(obj, key)) {
      size++
    }
  }

  return size
}
```

```
/home/travis/build/topaxi/learning-cs/utils/object/size.ts
  7:24  error  First argument must be name       jest/valid-describe
  7:24  error  Second argument must be function  jest/valid-describe
```

**Comment by SimenB** [comment]
what in the world...

**Comment by SimenB** [comment]
Oh, it's the `in DescribeAlias` things. It has a property called `hasOwnProperty`

**Comment by SimenB** [comment]
I'm trying to push a fix, but GH is having issues... https://www.githubstatus.com/

**Comment by topaxi** [comment]
Thanks again for the quick fixes ❤️
